// mjml-spacer/lib/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _mjmlCore = require("mjml-core");

class MjSpacer extends _mjmlCore.BodyComponent {
  getStyles() {
    return {
      div: {
        height: this.getAttribute('height'),
        'line-height': this.getAttribute('height')
      }
    };
  }

  render() {
    return `
      <div
        ${this.htmlAttributes({
      style: 'div'
    })}
      >&#8202;</div>
    `;
  }

}

exports.default = MjSpacer;
(0, _defineProperty2.default)(MjSpacer, "componentName", 'mj-spacer');
(0, _defineProperty2.default)(MjSpacer, "allowedAttributes", {
  border: 'string',
  'border-bottom': 'string',
  'border-left': 'string',
  'border-right': 'string',
  'border-top': 'string',
  'container-background-color': 'color',
  'padding-bottom': 'unit(px,%)',
  'padding-left': 'unit(px,%)',
  'padding-right': 'unit(px,%)',
  'padding-top': 'unit(px,%)',
  padding: 'unit(px,%){1,4}',
  height: 'unit(px,%)'
});
(0, _defineProperty2.default)(MjSpacer, "defaultAttributes", {
  height: '20px'
});
module.exports = exports.default;
