// mjml-wrapper/lib/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _mjmlSection = _interopRequireDefault(require("mjml-section"));

var _mjmlCore = require("mjml-core");

class MjWrapper extends _mjmlSection.default {
  renderWrappedChildren() {
    const {
      children
    } = this.props;
    const {
      containerWidth
    } = this.context;
    return `
      ${this.renderChildren(children, {
      renderer: component => component.constructor.isRawElement() ? component.render() : `
          <!--[if mso | IE]>
            <tr>
              <td
                ${component.htmlAttributes({
        align: component.getAttribute('align'),
        class: (0, _mjmlCore.suffixCssClasses)(component.getAttribute('css-class'), 'outlook'),
        width: containerWidth
      })}
              >
          <![endif]-->
            ${component.render()}
          <!--[if mso | IE]>
              </td>
            </tr>
          <![endif]-->
        `
    })}
    `;
  }

}

exports.default = MjWrapper;
(0, _defineProperty2.default)(MjWrapper, "componentName", 'mj-wrapper');
module.exports = exports.default;
