// mjml-body/lib/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _mjmlCore = require("mjml-core");

class MjBody extends _mjmlCore.BodyComponent {
  getChildContext() {
    return { ...this.context,
      containerWidth: this.getAttribute('width')
    };
  }

  getStyles() {
    return {
      div: {
        'background-color': this.getAttribute('background-color')
      }
    };
  }

  render() {
    const {
      setBackgroundColor
    } = this.context;
    setBackgroundColor(this.getAttribute('background-color'));
    return `
      <div
        ${this.htmlAttributes({
      class: this.getAttribute('css-class'),
      style: 'div'
    })}
      >
        ${this.renderChildren()}
      </div>
    `;
  }

}

exports.default = MjBody;
(0, _defineProperty2.default)(MjBody, "componentName", 'mj-body');
(0, _defineProperty2.default)(MjBody, "allowedAttributes", {
  width: 'unit(px)',
  'background-color': 'color'
});
(0, _defineProperty2.default)(MjBody, "defaultAttributes", {
  width: '600px'
});
module.exports = exports.default;
